packages/gatsby-transformer-mdx-introspection/README.md

```diff
@@ -40,31 +47,122 @@ query MyQuery {
 }
 ```
 
-Please not the lowercase `apitype` in the query - the names are normalized to lowercase by the DOM parser used.
-
-Example response:
+### Example response:
 
 ```json
 {
   "data": {
     "allComponentInMdx": {
       "nodes": [
         {
+          "component": "ApiType",
           "attributes": [
             {
-              "name": "apikey",
+              "name": "apiKey",
               "value": "test"
             },
             {
               "name": "type",
               "value": "OutOfOrderPropertiesTestType"
             }
           ],
-          "page": {
-            "fields": {
-              "slug": "/api/types/"
+          "mdx": {
+            "file": {
+              "relativePath": "api/types/index.md"
             }
           }
         },
-...
+        ...
+      ]
+    }
+  }
+}
+```
+
+## Options
+
+**`lowercaseIdentifiers`** [boolean] (optional, defaults to false)
+
+Whether to have the `component` field (and `attribute.name`) be all lower case (legacy)
+
+**`trimWhitespace`** [boolean] (optional, defaults to true)
+
+Whether to trim leading/trailing whitespace in JSX snippets and string literals
+
+**`collapseWhitespace`** [boolean] (optional, defaults to true)
+
+Whether to collapse consecutive whitespace in JSX snippets and string literals
+
+**`attachAST`** [boolean] (optional, defaults to false)
+
+Whether to attach the original JSX AST nodes to the generated node output. Needed to query `ComponentInMdx.ast` successfully
+
+**`removeMdxCompilationArtifacts`** [boolean] (optional, defaults to true)
+
+Whether to remove attributes that are (usually) artifacts of MDX compilation (`mdxType` and `parentName`)
+
+**`shouldIndexNode`** [(node) ⇒ boolean] (optional, defaults to () ⇒ true)
+
+Predicate function used as a performance escape hatch to filter MDX files that get parsed/indexed. Use if not all MDX files need to be indexed and build times are prohibitive
+
+**`excludeTags`** [array\<string>] (optional, defaults to [...])
```

Comment:
Since the defaults are inherently an assumption given specific sorts of markdown (which can contain arbitrary HTML) I wondered whether forcing an explicit whitelist wouldn't be better for the uses case we have in mind here (these all look for specific components that are hardwired in other parts of the implementation). 

I find whitelisting interesting because we have performance worries anyways and know use cases all look at a specific list of components or tags only.

Reply:
The whitelist approach is a good idea, and I think it serves most of the use cases I can think of, too. I would probably want to implement it using some `string | regular expression` syntax so that users could effectively implement their own blacklist with negative look-arounds.